http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/Queries.scala

```diff
@@ -0,0 +1,146 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package db
+
+import java.time.Instant
+
+import cats.data.Chain
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.api._
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.mappedColumnImplicits._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.ListClusterResponse
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, DataprocInfo}
+import org.broadinstitute.dsde.workbench.leonardo.model.google.{ClusterStatus, IP, OperationName}
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+
+import scala.concurrent.ExecutionContext
+
+object Queries {
+  type ClusterJoinLabelJoinRuntimeConfig = Query[((ClusterTable, Rep[Option[LabelTable]]), Rep[Option[RuntimeConfigTable]]), ((ClusterRecord, Option[LabelRecord]), Option[RuntimeConfigRecord]), Seq]
+
+  val runtimeConfigs = TableQuery[RuntimeConfigTable]
+
+  val clusterLabelRuntimeConfigQuery: ClusterJoinLabelJoinRuntimeConfig = {
+    for {
+      (cluster, label) <- clusterQuery
+        .joinLeft(labelQuery).on(_.id === _.clusterId)
+        .joinLeft(runtimeConfigs).on(_._1.runtimeConfigId === _.id)
+    } yield (cluster, label)
+  }
+
+  /**
+    * return DB generated id
+   */
+  def upsertRuntime(runtimeConfig: RuntimeConfig, dateAccessed: Instant): DBIO[RuntimeConfigId] =  {
+    (runtimeConfigs.returning(runtimeConfigs.map(_.id)) += RuntimeConfigRecord(RuntimeConfigId(0), runtimeConfig, dateAccessed))
+  }
+
+  def getRuntime(id: RuntimeConfigId)(implicit ec: ExecutionContext): DBIO[RuntimeConfig] =  {
+    runtimeConfigs.filter(x => x.id === id).result.flatMap {
+      x =>
+        val res = x.headOption.map {
+          x =>
+            x.runtimeConfig
+        }
+        res.fold[DBIO[RuntimeConfig]](DBIO.failed(new Exception(s"no runtimeConfig found for ${id}")))(x => DBIO.successful(x))
+    }
+  }
+
+  def getClustersByLabelsWithRuntimeConfig(labelMap: LabelMap, includeDeleted: Boolean, googleProjectOpt: Option[GoogleProject] = None)(implicit ec: ExecutionContext): DBIO[List[ListClusterResponse]] = {
```

Comment:
Could the code in ClusterComponent `listByLabelsQuery` be replaced with this?

Reply:
yea...Ill remove `listByLabelsQuery`